grafana/grafana-infinity-datasource
# dependencies update

## 2.5.0

🛡️ **Security** Backend binaries compiled with go 1.22.1 which includes [security patches](https://groups.google.com/g/golang-announce/c/5pwGVUPoMbg)

⚙️ **Dependencies update**: NodeJS dev dependency updated from Node 16 to Node 20

⚙️ **Dependencies update**: Updated backend dependencies and plugin SDK update

⚙️ **Chore**: Replaced gorilla mux with server mux from standard library

⚙️ **Chore**: Removed unused graphql and open-api resource calls

### How to test

There is no much functionality change. dependencies updated. To test the mux migration, run the following commonds

```sh
yarn 
yarn build
mage -v
docker compose up
```
and then once the grafana is up, [this url](http://localhost:3000/api/datasources/uid/Infinity/resources/reference-data) should show the list of reference data array and this [explore URL](http://localhost:3000/explore?schemaVersion=1&panes=%7B%223Kp%22:%7B%22datasource%22:%22Infinity%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22yesoreyeram-infinity-datasource%22,%22uid%22:%22Infinity%22%7D,%22type%22:%22tsv%22,%22source%22:%22reference%22,%22format%22:%22table%22,%22url%22:%22https:%2F%2Fgithub.com%2Fgrafana%2Fgrafana-infinity-datasource%2Fblob%2Fmain%2Ftestdata%2Fusers.csv%22,%22url_options%22:%7B%22method%22:%22GET%22,%22data%22:%22%22%7D,%22root_selector%22:%22%22,%22columns%22:%5B%5D,%22filters%22:%5B%5D,%22global_query_id%22:%22%22,%22referenceName%22:%22users.tsv%22,%22parser%22:%22backend%22%7D%5D,%22range%22:%7B%22from%22:%22now-6h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1) should produce some results

### Note

CodeQL backend checks are failing due to https://github.com/github/codeql/issues/15647
